logic/account.js

```diff
@@ -580,13 +582,33 @@ Account.prototype.getAll = function (filter, fields, cb) {
 	delete filter.offset;
 
 	if (filter.sort) {
-		sort = filter.sort;
+		sort = orderBy.sortQueryToJsonSqlFormat(filter.sort, ['username', 'balance']);
 	}
 	delete filter.sort;
 
-	if (typeof filter.address === 'string') {
-		filter.address = {
-			$upper: ['address', filter.address]
+	if (filter.address) {
+
+		if (typeof filter.address === 'string') {
+			filter['a.address'] = {
+				$upper: ['a.address', filter.address]
+			};
+		}
+
+		if (typeof filter.address === 'object') {
+			filter['a.address'] = filter.address;
+		}
+		delete filter.address;
+	}
+
+	if (typeof filter.publicKey === 'string') {
```

Comment:
For other fields we are checking if the field just exists

Reply:
That is because publicKey is a common field b/w delegates table and accounts table. And we need to add alias for fields which are common among table.